internal/x/max_depth.go

```diff
@@ -0,0 +1,23 @@
+package x
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+)
+
+func GetMaxDepthFromQuery(q url.Values, required bool) (int, error) {
+	if !q.Has("max-depth") {
+		if required {
+			return 0, fmt.Errorf("required query parameter 'max-depth' is missing")
```

Comment:
We usually return `errors.WithStack` as a convention. Also, you could make these already `herodot.ErrBadRequest.WithError()`, so that the response status codes are set properly. All errors are bad requests, so it does not matter too much, but it would allow the handlers to directly write the error. In case there will be a new case at some point, it would also allow the use of different status codes.

Reply:
For this I thought `GetMaxDepthFromQuery` might be independent of considerations about the form the response takes, this would allow the caller to make those decisions.

Reply:
My bad, it is not anymore possible. Forgot that.